src/main/java/io/trino/tpch/SharingTextPoolSupplier.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tpch;
+
+import javax.annotation.concurrent.GuardedBy;
+
+import java.lang.ref.WeakReference;
+
+import static java.util.Objects.requireNonNull;
+
+public final class SharingTextPoolSupplier
+        implements java.util.function.Supplier<TextPool>
```

Comment:
Why not import it?

Reply:
IntelliJ got confused by existence of `Supplier` class in this package, and I got confused together with it.